File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 6/10/2026, 11:51:32 AM
public class Solution {
    public int lengthOfLongestSubstring(String s) {
        String temp = "" ,help = "";
        int max = 0;//"jbpnbwwd"
        if(s.length() == 1) return (1);
        for(int i = 0 ; i < s.length() - 1; i++)
                for (int j = i ; j < s.length() ; j++)
                    {   help += s.charAt(j);
                        if(!temp.contains(help))
                            temp +=  s.charAt(j);
                        else
                        {
                            temp = "";
                            break;
                        }
                        if(max < temp.length())
                            max = temp.length();
                        help = "";
                    }
        return (max);
    }
}

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 6/10/2026, 11:51:30 AM
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        // string s; cin>>s;
    int n = s.size();
    set<char> x;
    int mx = 0; 
    for(int i = 0 , j = 0; i < n;  i++){
        if(x.find(s[i]) == x.end()){
            x.insert(s[i]);
        }
        else{
            while(x.find(s[i]) != x.end()){
                x.erase(s[j]);
                j++;
            }
            x.insert(s[i]);
        }
        int len = i-j+1;
        mx = max(mx , len);
    }
    cout<<mx;
    return mx;
    }
};